core/systems/promotion/form_view.ex

```diff
@@ -132,6 +134,48 @@ defmodule Systems.Promotion.FormView do
     end
   end
 
+  def update_wysiwyg_form(
+        %{assigns: %{entity: %{expectations: expectations, description: description}}} = socket
+      ) do
+    wysiwyg_form =
+      to_form(%{
+        "expectations" => expectations || "",
+        "description" => description || ""
+      })
+
+    socket |> assign(wysiwyg_form: wysiwyg_form)
+  end
+
+  @impl true
+  def handle_wysiwyg_update(%{assigns: assigns} = socket) do
```

Comment:
Consider 3 `handle_wysiwyg_update` functions with different pattern matching on `expectations` and `description` iso the 'cond do'. Nothing inherently wrong with your solution though function head pattern matching is a more declarative programming approach.